pkg/batch/jobs/calculator/experiment_calculate.go

```diff
@@ -91,10 +93,19 @@ func NewExperimentCalculate(
 }
 
 func (e *experimentCalculate) Run(ctx context.Context) error {
+	// Prevent goroutine stacking by checking if a calculation is already running
+	if !e.running.CompareAndSwap(false, true) {
+		e.logger.Warn("Experiment calculation already in progress, skipping this cycle",
+			zap.String("reason", "previous calculation not completed"),
+		)
+		return nil
+	}
```

Comment:
Here, we prevent goroutine stacking when receiving requests from cron jobs while the calculation is still in progress.
In case of other pods getting the request, it will run the calculation normally, but it will still check the Redis Lock to avoid calculating the same experiment across pods.